api/services/cts.go

```
@@ -17,12 +16,12 @@ type ServiceClient struct {
 	taskConn    *grpc.ClientConn
 }
 
-func NewClient(addr string) *ServiceClient {
+func NewClient(addr string) (*ServiceClient, error) {
 	var opts []grpc.DialOption
 	opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	taskConn, err := grpc.Dial(addr, opts...)
 	if err != nil {
-		log.Fatalf("fail to dial: %v", err)
+		return nil, err
```

why fail silently now?

---

I have a retry on the helper